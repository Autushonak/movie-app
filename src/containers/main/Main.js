import React, { useEffect } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { createSelector } from 'reselect';
import { connect } from 'react-redux';
import { colors } from '../../constants/colors';
import ControllerBar from '../controller/ControllerBar';
import MovieList from '../movie-list/MovieList';
import { EmptyState } from '../../components';
import { getMovies, changeSortBy } from '../../store/actions/movie';

const MainStyled = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  min-height: 400px;
  background-color: ${colors.GREY_MAIN};
`;

const Content = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  min-height: 300px;
`;

const Main = ({ total, movies, sortBy, getMovies, changeSortBy, suggestedGenre, mode }) => {
  useEffect(() => {
    getMovies();
  }, []);
  const handleSortByChange = (sortBy) => {
    changeSortBy(sortBy);
    getMovies();
  };
  return (
    <MainStyled>
      <ControllerBar
        total={total}
        sortBy={sortBy}
        handleSortByChange={handleSortByChange}
        suggestedGenre={suggestedGenre}
        mode={mode}
      />
      <Content>
        {movies && !!movies.length ? <MovieList movies={movies} /> : <EmptyState>No films found</EmptyState>}
      </Content>
    </MainStyled>
  );
};

Main.propTypes = {
  total: PropTypes.number,
  movies: PropTypes.array,
  sortBy: PropTypes.string,
  getMovies: PropTypes.func,
  changeSortBy: PropTypes.func,
  suggestedGenre: PropTypes.string,
  mode: PropTypes.string,
};

const itemsSelector = createSelector(
  (state) => state.movie.movieList,
  (items) => items.toArray(),
);

const mapStateToProps = (state) => ({
  movies: itemsSelector(state),
  sortBy: state.movie.sortBy,
  total: state.movie.total,
  suggestedGenre: state.movie.suggestedGenre,
});

const mapDispatchToProps = {
  getMovies,
  changeSortBy,
};

export default connect(mapStateToProps, mapDispatchToProps)(Main);
export { MainStyled, Content, Main };
