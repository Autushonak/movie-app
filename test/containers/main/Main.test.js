import 'jest-styled-components';
import React from 'react';
import { shallow } from 'enzyme';
import Main, { MainStyled, Content } from '../../../src/containers/main/Main';

const movies = [
  {
    url: 'https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg',
    title: 'Guardians of the Galaxy Vol. 3',
    year: '2020',
    genre: 'Science Fiction',
  },
];

describe('MainStyled', () => {
  test('should be rendered correctly', () => {
    const component = shallow(<MainStyled height={500} />);
    expect(component).toMatchSnapshot();
  });
});

describe('Content', () => {
  test('should be rendered correctly', () => {
    const component = shallow(<Content />);
    expect(component).toMatchSnapshot();
  });
});

describe('Main', () => {
  test('should be rendered without movies', () => {
    const component = shallow(<Main />);
    expect(component).toMatchSnapshot();
  });

  test('should be rendered with movies', () => {
    const component = shallow(<Main movies={movies} />);
    expect(component).toMatchSnapshot();
  });
});
