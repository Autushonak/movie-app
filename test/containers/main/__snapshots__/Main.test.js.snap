// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should be rendered correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 300px;
}

<div
  className="c0"
/>
`;

exports[`Main should be rendered with movies 1`] = `
<styled.div>
  <ControllerBar />
  <styled.div>
    <MovieList
      movies={
        Array [
          Object {
            "genre": "Science Fiction",
            "title": "Guardians of the Galaxy Vol. 3",
            "url": "https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg",
            "year": "2020",
          },
        ]
      }
    />
  </styled.div>
</styled.div>
`;

exports[`Main should be rendered without movies 1`] = `
<styled.div>
  <ControllerBar />
  <styled.div>
    <styled.h1>
      No films found
    </styled.h1>
  </styled.div>
</styled.div>
`;

exports[`MainStyled should be rendered correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  min-height: 400px;
  background-color: #232323;
}

<div
  className="c0"
  height={500}
/>
`;
